artnet/__init__.py

```diff
@@ -21,36 +22,170 @@
+
+
+class ArtNet_Model:
+
+    def __init__(self, ip, port, net, subnet, universe, instance_name, update_cycle):
+        self._ip = ip
+        self._port = port
+
+        self._universe = universe
+        self._net = net
+        self._subnet = subnet
+        self._instance_name = instance_name
+        self._update_cycle = update_cycle
+
+        self._items = []
+
+    def get_ip(self):
+        """
+        Returns the IP of the ArtNet node
+
+        :return: IP-address of the device, as set in plugin.conf
+        """
+        return self._ip
+
+    def get_port(self):
+        """
+        Returns the port of the ArtNet node
+
+        :return: port of the device, as set in plugin.conf
+        """
+        return self._port
+
+    def get_net(self):
+        """
+        Returns the net of the ArtNet node
+
+        :return: net of the device, as set in plugin.conf
+        """
+        return self._net
+
+    def get_subnet(self):
+        """
+        Returns the Subnet of the ArtNet node
+
+        :return: Subnet of the device, as set in plugin.conf
+        """
+        return self._subnet
+
+    def get_universe(self):
+        """
+        Returns the Universe of the ArtNet node
+
+        :return: Universe of the device, as set in plugin.conf
+        """
+        return self._universe
+
+    def get_items(self):
+        """
+        Returns added items
+
+        :return: array of items held by the device
+        """
+        return self._items
 
 
 class ArtNet(SmartPlugin):
 
-    ALLOW_MULTIINSTANCE = False
-    PLUGIN_VERSION = '1.3.0'
+    ALLOW_MULTIINSTANCE = True
+    PLUGIN_VERSION = "1.6.0"
+    ADDR_ATTR = 'artnet_address'
 
     packet_counter = 1
     dmxdata = [0, 0]
 
-    def __init__(self, smarthome, artnet_net, artnet_subnet, artnet_universe, ip, port):
-        self.logger = logging.getLogger(__name__)
+    def __init__(self, sh, *args, **kwargs):
+        """
+        Initalizes the plugin. The parameters describe for this method are pulled from the entry in plugin.conf.
+        """
+        self.logger.info('Init ArtNet Plugin')
+
+        self._model = ArtNet_Model(self.get_parameter_value('ip') or '127.0.0.1',
+                                   port=int(self.get_parameter_value(
+                                       'port') or 6454),
+                                   net=int(self.get_parameter_value(
+                                       'artnet_net') or 0),
+                                   subnet=int(self.get_parameter_value(
+                                       'artnet_subnet') or 0),
+                                   universe=int(self.get_parameter_value(
+                                       'artnet_universe') or 0),
+                                   instance_name=self.get_instance_name(),
+                                   update_cycle=int(
+                                       self.get_parameter_value('update_cycle') or 0),
+                                   )
```

Comment:
All these `get_parameter_value(...) or XY` are use to get the parameter value or a default value? The default value for a parameter can also be specified in the `plugin.yml`.

Reply:
Yes but this is neither in effect if you change the installations plugin.yaml manually nor if you have shng151, right? I added the defaults to the plugin.yaml (of this plugin) now.

Reply:
You are wrong. Since shng 1.4 the get_parameter_value(...) returns the default value, if no value is configured for the parameter in /etc/plugin.yaml. You can even simplify it further. You don't need to convert the values to int. get_parameter_value(...) returns the value in the datatype that corresponds to the type defined in plugin.yaml (as described here: https://www.smarthomeng.de/developer/development_plugin/smartplugin.html?highlight=get_parameter_value#lib.model.smartplugin.SmartPlugin.get_parameter_value)

Reply:
fixed it accordingly